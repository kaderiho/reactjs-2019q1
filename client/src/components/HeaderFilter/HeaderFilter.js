import React, { Component }  from 'react';

import Button from 'components/Button/Button';
import Input from 'components/Input/Input';
import './HeaderFilter.scss';

class HeaderFilter extends Component {
    state = {
        filterBy: 'filterByTitle',
        searchFilter: ''
    }

    // On movie search handler
    onInputHandler = ev => {
        this.setState({
            [ev.target.name]: ev.target.value
        });
    }

    // On filter by controls handler
    onFilterByHandler = ev => {
        this.setState({
            filterBy: ev.target.name
        });
    }

    // On search handler
    onSearchHandler = () => {console.log('Search!')}

    render() {
        let { filterBy } = this.state;

        return (
            <div className="headerFilter">
                <p className="headerFilter-title">Find your movie</p>
                <Input 
                    placeholder="Type movie name here"
                    value={this.state.searchFilter}
                    onChange={this.onInputHandler}
                    name="searchFilter"
                    id="searchFilter"
                />

                <div className="headerSubFilter">
                    <span className="headerSubFilter-label">Search by</span>
                    <Button className="headerSubFilter-button"
                            onClick={this.onFilterByHandler}
                            name="filterByTitle"
                            color={filterBy === 'filterByTitle' ? 'secondary' : 'third'}
                            size="small">
                            Title
                    </Button>
                    <Button className="headerSubFilter-button"
                            onClick={this.onFilterByHandler}
                            name="filterByGenre"
                            color={filterBy === 'filterByGenre' ? 'secondary' : 'third'}
                            size="small">
                            Genre
                    </Button>
                    <Button className="headerSubFilter-searchButton"
                            onClick={this.onSearchHandler}
                            color="secondary">
                            Search
                    </Button>
                </div>
            </div>
        )
    }
};

export default HeaderFilter;