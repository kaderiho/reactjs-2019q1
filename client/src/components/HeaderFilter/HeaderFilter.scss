.headerFilter {
    padding: 30px 32px;
}

.headerSubFilter {
    display: flex;
    align-items: center;
    margin-top: 20px;
}

.headerSubFilter-button + .headerSubFilter-button {
    margin-left: 10px;
}
