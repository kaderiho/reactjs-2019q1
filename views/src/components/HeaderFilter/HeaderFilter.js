"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.HeaderFilter = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _reactRouterDom = require("react-router-dom");

var _queryString = _interopRequireDefault(require("query-string"));

var _Button = _interopRequireDefault(require("../Button/Button"));

var _Input = _interopRequireDefault(require("../Input/Input"));

var HeaderFilter =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(HeaderFilter, _Component);

  function HeaderFilter() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, HeaderFilter);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(HeaderFilter)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      searchStr: '',
      searchBy: ''
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onInputHandler", function (ev) {
      _this.setState({
        searchStr: ev.target.value
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onFilterByHandler", function (ev) {
      _this.setState({
        searchBy: ev.target.name
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "onSearchHandler", function (e) {
      e.preventDefault();
      var _this$state = _this.state,
          searchStr = _this$state.searchStr,
          searchBy = _this$state.searchBy;
      var search = {};

      if (searchStr) {
        search['searchStr'] = searchStr;
      }

      if (searchBy) {
        search['searchBy'] = searchBy;
      }

      if (searchStr || searchBy) {
        _this.props.history.push({
          pathname: '/search/',
          search: "?".concat(_queryString.default.stringify(search))
        });
      }
    });
    return _this;
  }

  (0, _createClass2.default)(HeaderFilter, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _queryString$parse = _queryString.default.parse(this.props.location.search),
          searchStr = _queryString$parse.searchStr,
          searchBy = _queryString$parse.searchBy;

      this.setState({
        searchStr: searchStr || '',
        searchBy: searchBy || ''
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$state2 = this.state,
          searchStr = _this$state2.searchStr,
          searchBy = _this$state2.searchBy;
      return _react.default.createElement("div", {
        className: "headerFilter"
      }, _react.default.createElement("form", {
        onSubmit: this.onSearchHandler
      }, _react.default.createElement("p", {
        className: "headerFilter-title"
      }, "Find your movie"), _react.default.createElement(_Input.default, {
        placeholder: "Type movie name here",
        value: searchStr,
        onChange: this.onInputHandler,
        name: "searchFilter",
        id: "searchFilter"
      }), _react.default.createElement("div", {
        className: "headerSubFilter"
      }, _react.default.createElement("span", {
        className: "headerSubFilter-label"
      }, "Search by"), _react.default.createElement(_Button.default, {
        className: "headerSubFilter-button",
        onClick: this.onFilterByHandler,
        name: "title",
        color: searchBy === 'title' ? 'secondary' : 'third',
        size: "small"
      }, "Title"), _react.default.createElement(_Button.default, {
        className: "headerSubFilter-button",
        onClick: this.onFilterByHandler,
        name: "genres",
        color: searchBy === 'genres' ? 'secondary' : 'third',
        size: "small"
      }, "Genre"), _react.default.createElement(_Button.default, {
        className: "headerSubFilter-searchButton",
        onClick: this.onSearchHandler,
        color: "secondary",
        type: "submit"
      }, "Search"))));
    }
  }]);
  return HeaderFilter;
}(_react.Component);

exports.HeaderFilter = HeaderFilter;
;

var _default = (0, _reactRouterDom.withRouter)(HeaderFilter);

exports.default = _default;