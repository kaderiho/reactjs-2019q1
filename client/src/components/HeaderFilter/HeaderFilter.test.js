import React from 'react';
import { shallow, mount } from 'enzyme';

import HeaderFilter from './HeaderFilter';

/**
 * Tests
 * 
 * - To match snapshot
 * - onFilterByHandler should be called if button is clicked
 * - onInputHandler should be called if input is changed
 * - Input value is changed correctly
 */

describe('<HeaderFilter /> component', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<HeaderFilter />);
        
        expect(wrapper).toMatchSnapshot();
    });

    it('onFilterByHandler is called', () => {
        const wrapper = mount(<HeaderFilter />);
        const spy = jest.spyOn(wrapper.instance(), 'onFilterByHandler');
        wrapper.instance().forceUpdate();
        
        const filterButton = wrapper.find('button[name="FilterByTitle"]');
        filterButton.simulate('click');
        expect(spy).toBeCalled();
    });

    it('onInputHandler is called', () => {
        const wrapper = mount(<HeaderFilter />);
        const spy = jest.spyOn(wrapper.instance(), 'onInputHandler');
        wrapper.instance().forceUpdate();

        const filterInput = wrapper.find('input[name="searchFilter"]');
        filterInput.simulate('change', { target: { value: 'My new value' }});
        expect(spy).toBeCalled();
    });

    it('Input value is changed', () => {
        const wrapper = mount(<HeaderFilter />);

        const filterInput = wrapper.find('input[name="searchFilter"]');
        filterInput.instance().value = "My new value"
        filterInput.simulate('change');

        expect(wrapper.find('Input').prop('value')).toBe('My new value');
    });
});