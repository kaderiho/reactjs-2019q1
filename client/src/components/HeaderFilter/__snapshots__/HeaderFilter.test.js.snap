// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderFilter /> component Should render correctly 1`] = `
<div
  className="headerFilter"
>
  <p
    className="headerFilter-title"
  >
    Find your movie
  </p>
  <Input
    id="searchFilter"
    name="searchFilter"
    onChange={[Function]}
    placeholder="Type movie name here"
    value=""
  />
  <div
    className="headerSubFilter"
  >
    <span
      className="headerSubFilter-label"
    >
      Search by
    </span>
    <Button
      className="headerSubFilter-button"
      color="secondary"
      name="filterByTitle"
      onClick={[Function]}
      size="small"
    >
      Title
    </Button>
    <Button
      className="headerSubFilter-button"
      color="third"
      name="filterByGenre"
      onClick={[Function]}
      size="small"
    >
      Genre
    </Button>
    <Button
      className="headerSubFilter-searchButton"
      color="secondary"
      onClick={[Function]}
    >
      Search
    </Button>
  </div>
</div>
`;
